src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -235,21 +235,33 @@ missingAlternative env mn ca uncovered
 
 -- |
 -- Main exhaustivity checking function
--- Starting with the set `uncovered = { _ }` (nothing covered, one `_` for each function argument),
--- it partitions that set with the new uncovered cases, until it consumes the whole set of clauses.
--- Then, returns the uncovered set of case alternatives.
+-- If there are no cases, check that at least one argument is of an empty type.
+-- Otherwise, starting with the set `uncovered = { _ }` (nothing covered, one
+-- `_` for each function argument), it partitions that set with the new
+-- uncovered cases, until it consumes the whole set of clauses. Then, returns
+-- the uncovered set of case alternatives.
 --
 checkExhaustive
   :: forall m
    . (MonadWriter MultipleErrors m, MonadSupply m)
    => Environment
    -> ModuleName
-   -> Int
+   -> [Expr]
    -> [CaseAlternative]
    -> Expr
    -> m Expr
-checkExhaustive env mn numArgs cas expr = makeResult . first ordNub $ foldl' step ([initialize numArgs], (pure True, [])) cas
+checkExhaustive env mn args cas expr =
+  makeResult . first ordNub $
+    if null cas
```

Comment:
I am not familiar with the exhaustivity checker, I'll have to spend some more time on it.

I don't think we want to signal that as redundant, because then you have no way of propagating the `Void` proof to something else anymore.